tools/clone_fileset/main/main.go

```diff
@@ -0,0 +1,98 @@
+package main
+
+import (
+	"flag"
+	"io"
+	"os"
+
+	"github.com/m3db/m3db/persist/encoding/msgpack"
+	"github.com/m3db/m3db/persist/fs"
+	"github.com/m3db/m3db/ts"
+	xlog "github.com/m3db/m3x/log"
+	"github.com/m3db/m3x/pool"
+	xtime "github.com/m3db/m3x/time"
+)
+
+const (
+	defaultBufferSize      = 4096
+	defaultBufferCapacity  = 256
+	defaultBufferPoolCount = 3145728
+)
+
+var (
+	optSrcPathPrefix  = flag.String("src-path-prefix", "/var/lib/m3db", "Source Path prefix")
+	optSrcNamespace   = flag.String("src-namespace", "metrics", "Source Namespace")
+	optSrcShard       = flag.Uint("src-shard-id", 0, "Source Shard ID")
+	optSrcBlockstart  = flag.Int64("src-block-start", 0, "Source Block Start Time [in nsec]")
+	optDestPathPrefix = flag.String("dest-path-prefix", "/tmp/m3db-copy", "Destination Path prefix")
+	optDestNamespace  = flag.String("dest-namespace", "metrics", "Destination Namespace")
+	optDestShard      = flag.Uint("dest-shard-id", 0, "Destination Shard ID")
+	optDestBlockstart = flag.Int64("dest-block-start", 0, "Destination Block Start Time [in nsec]")
+	optDestBlockSize  = flag.Duration("dest-block-size", 0, "Destination Block Size")
+)
+
+var (
+	targetFileMode = os.FileMode(0666)
+	targetDirMode  = os.ModeDir | os.FileMode(0755)
+	log            = xlog.NewLogger(os.Stderr)
+)
+
+func main() {
+	flag.Parse()
+	if *optSrcPathPrefix == "" ||
+		*optDestPathPrefix == "" ||
+		*optSrcNamespace == "" ||
+		*optDestNamespace == "" ||
+		*optSrcBlockstart <= 0 ||
+		*optDestBlockstart <= 0 {
+		flag.Usage()
+		os.Exit(1)
+	}
+
+	bytesPool := pool.NewCheckedBytesPool([]pool.Bucket{pool.Bucket{
```

Comment:
Hey as discussed, found for this case no pooling is roughly 3x faster (probably because GC never has enough time to kick in during execution as its less than a few seconds) .

Let's kick the pooling out?